### PR TITLE
Allow toggling with a swipe gesture, fixes #49

### LINKED ISSUE #49: Allow dragging toggle
Some people try to use the toggles by dragging, but the action fails since we don't support that.

### PR DESCRIPTION
**Pull request by lolmaus**
Implemented toggling with a swipe gesture. Fixes #49.

Notes:

1. I used [ember-gestures](https://github.com/html-next/ember-gestures) for finger gestures support.
2. I used simple panning instead of actual handle dragging because the logic behind dragging would be *way* more complicated. You can have an idea from [this prototype](https://github.com/html-next/ember-gestures/blob/v0.4.5/addon/components/slide-toggle.js). The gotcha is that with panning toggling happens immediately and not after mouse/finger release. 
3. `ember-gestures` needs an update of `ember-hammertime` to support Ember 2.15+. [PR pending](https://github.com/html-next/ember-gestures/pull/100). This PR currently uses a feature branch of `ember-gestures`. You can try it out locally, but don't merge this yet.
4. Animation direction of the `skewed` styles inverted for consistency with all other styles. This makes a difference because swiping is directional. 
5. Fixed a bug with `skewed` active state: on Chrome Desktop with mouse clicking it didn't work, on Chrome Android with finger tapping/swiping the toggle stayed active (and misaligned) indefinitely, until user tapped elsewhere.

### CODE REVIEW

**Comment by lolmaus** [comment]
Something weird in Travis. Can you please try clearing all caches and restarting the build?

**Comment by knownasilya** [comment]
Cleared and restarted

**Comment by eriktrom** [comment]
fyi - blocked by https://github.com/html-next/ember-gestures/pull/102 fyi

**Comment by lolmaus** [comment]
@vvainio has brought my attention to the fact that my initial implementation does not support mouse gestures, only touch. This was probably due to the fact that `ember-gestures` has a minimum stroke length for the swipe gesture which is rarely achieved with a mouse.

I've changed `swipe` to `pan`, so now both mouse and finger gestures behave consistently.

This revealed a gotcha: when you pan with a mouse and release the mouse button over the toggle, a click even is produced on the `<label>` element, which immediately undoes toggling. I came up with a witty workaround which resolves the problem without interfering with a11y.

@eriktrom I've bumped Node and it didn't help. Lets wait for https://github.com/html-next/ember-gestures/pull/102 to finalize, it may resolve the issue.

**Comment by webark** [comment]
@lolmaus I'm currently using this in a project without jquery. Is there anyway we could handle this without having to use it?

**Comment by lolmaus** [comment]
@webark Granted.

**Comment by webark** [comment]
🙏

**Comment by eriktrom** [comment]
ember-gestures fixed upstream, released as 0.4.7, fyi

**Comment by lolmaus** [comment]
@eriktrom Thank you. :bow: PR updated.

File: addon/components/x-toggle-switch/component.js [hunk 1]
```diff
@@ -1,14 +1,51 @@
...
+  /*
+    When you pan with a mouse and release the mouse button over the <label>
+    element, a click event happens and returns the toggle to its initial
+    state. :(
+
+    To prevent this, we need to make the label non-functional until after the
+    mouse button is released.
+   */
+  _disableLabelUntilMouseUp () {
+    if (this.get('labelDisabled')) return;
```
Comment (knownasilya) [inline addon/components/x-toggle-switch/component.js]: Could you make this a multi line if with `{ .. }`.

File: addon/components/x-toggle-switch/component.js [hunk 2]
```diff
@@ -1,14 +1,51 @@
...
+
+    this.set('labelDisabled', true);
+
+    document.addEventListener('mouseup', () => {
+      next(() => this.set('labelDisabled', false));
+    })
```
Comment (knownasilya) [inline addon/components/x-toggle-switch/component.js]: Missing semicolon

**Comment by lolmaus** [comment]
@knownasilya I've addressed your comments and squashed.

**Comment by knownasilya** [comment]
Published as 5.1.0

**Comment by lolmaus** [comment]
@knownasilya Thank you, comrade.

**Pull request merged**